src/databricks/labs/ucx/framework/parallel.py

```diff
@@ -45,22 +48,36 @@ def __init__(
     @classmethod
     def gather(cls, name: str, tasks: list[ExecutableFunction]) -> list[ExecutableResult]:
         reporter = ProgressReporter(len(tasks), f"{name}: ")
-        return cls(tasks, num_threads=4, progress_reporter=reporter).run()
+        return cls(name, tasks, num_threads=4, progress_reporter=reporter).run()
 
     def run(self) -> list[ExecutableResult]:
         logger.debug(f"Starting {len(self._executables)} tasks in {self._num_threads} threads")
 
+        def error_handler(func, name):
+            @functools.wraps(func)
+            def inner(*args, **kwargs):
+                try:
+                    return func(*args, **kwargs)
+                except Exception as ex:
+                    msg = getattr(ex, "message", ex)
+                    logger.warning(f"An error occurred while '{name}', message: {msg}")
```

Comment:
```suggestion
                    logger.warning(f"{name} task failed: {str(ex)}")
```
python exceptions don't always have `message`. and we need more concise log messages.

Reply:
But that is why I have the `getattr()` with the default value. Why call `str()` on the exception, a simple `{ex}` does the same, no (that is, call `__str__`)?